### ISSUE TITLE
Logger is undefined while loading hawtio-core.js

### ISSUE THREAD

**Issue by BoykoAlex** [hawtio/hawtio-core#2]
I'm trying to use just the hawtio-core.js as a bower dependency for loading plugins dynamically. Plugin would be a JS loading an angular module for example.
My project is using require js as well. Hence, in the main.js I call a require on hawtio-core js which depends on `angular, angular-sanitize, jquery, js-logger`. The JS files are found and loaded including the `js-logger.js`, but then when `hawtio-core.js` loads it fails right away on first lines initializing the Logger object. Logger is undefined. Any idea what is going wrong? I went over the `js-logger.js` and wonder if I need to do anything special to have `Logger` as a global variable...
Thanks!

**Comment by gashcrumb** [comment]
We haven't really done any integration with require.js in hawtio, so there definitely could be some issues.  Wonder if something like this would work:

```
window.Logger = require('js-logger');  // basically just ensure that Logger is global before loading the core
var HawtioCore = require('hawtio-core');
```

as all of hawtio assumes that 'Logger' is available in the global namespace.  Let me know how it goes!

**Comment by BoykoAlex** [comment]
That helped, thanks very much. (I hacked the js-logger yesterday to assign the global Logger variable regardless of Require JS presence. It worked, but your proposed fix I like much better ;-))

Now, however, it looks like HawtioCore is loading plugins (performing bootstrapping) before I add my plugins via `hawtioPluginLoader.addUrl(...)`. I've tried calling `hawtioPluginLoader.loadPlugins()` and it seemed to execute without errors... I went through it with the debugger and it seems like it gets the JSON data with the scripts from the URL, finds the scripts and attempts to load them... The script loading promise get the right text... The page however seems not to be bootstrapped with the angular app.

Is what I'm doing above make sense?
